crypto/asn1/x_int64.c

```diff
@@ -28,13 +28,21 @@
 
 static int uint64_new(ASN1_VALUE **pval, const ASN1_ITEM *it)
 {
-    *(uint64_t *)pval = 0;
+    *pval = (ASN1_VALUE *)OPENSSL_zalloc(sizeof(uint64_t));
+    if (*pval == NULL)
+        return 0;
     return 1;
 }
 
 static void uint64_free(ASN1_VALUE **pval, const ASN1_ITEM *it)
 {
-    *(uint64_t *)pval = 0;
+    OPENSSL_free(*pval);
+    *pval = NULL;
+}
+
+static void uint64_clear(ASN1_VALUE **pval, const ASN1_ITEM *it)
+{
+    **(uint64_t **)pval = 0;
```

Comment:
If you have a look around, you'll find that `ASN1_VALUE` has no definition apart from a typedef:

```
: ; git grep ASN1_VALUE_st
include/openssl/asn1.h:typedef struct ASN1_VALUE_st ASN1_VALUE;
```

That means that `*pval` is create as an object of *whatever* and then being cast to `ASN1_VALUE *` so we have a consistent C type to throw around in the rest of the code.

It's true, though, that we "play a little fast-and-loose" insofar that we trust the programmer to map the ASN.1 types to the correct corresponding C type.  If the programmer gets that wrong, there's hell to pay (quite literally in some cases).  This is well known, and we dont worry about it...  too much ;-)  (that, or we quietly wisper "Here be dragons" and hope we don't awaken C'Thulu)